tools/serverpod_cli/test/integration/analyzer/dart/endpoint_validation/valid_endpoints_test.dart

```diff
@@ -0,0 +1,808 @@
+      endpointFile.createSync(recursive: true);
+      endpointFile.writeAsStringSync('''
+import 'package:serverpod/serverpod.dart';
+
+class ExampleEndpoint extends Endpoint {
+  Future<String> hello(Session session, String name, {required String named}) async {
+    return 'Hello \$name';
+  }
+}
+''');
+      analyzer = EndpointsAnalyzer(testDirectory);
+      endpointDefinitions = await analyzer.analyze(collector: collector);
+    });
+
+    test('then no parsing errors are reported.', () {
+      expect(analyzer.getErrors(), completion(isEmpty));
+    });
+
+    test('then no validation errors are reported.', () {
+      expect(collector.errors, isEmpty);
+    });
+
+    test('then endpoint definition is created.', () {
+      expect(endpointDefinitions, hasLength(1));
+    });
+
+    test('then endpoint definition has one method defined.', () {
+      var methods = endpointDefinitions.firstOrNull?.methods;
+      expect(methods, hasLength(1));
+    });
+
+    test('then endpoint definition method has one positional parameter.', () {
+      var parametersNamed =
+          endpointDefinitions.firstOrNull?.methods.firstOrNull?.parametersNamed;
+      expect(parametersNamed, hasLength(1));
+    });
+
+    test('then endpoint methods positional parameter has correct name', () {
+      var name = endpointDefinitions
+          .firstOrNull?.methods.firstOrNull?.parametersNamed.firstOrNull?.name;
+      expect(name, 'named');
+    });
+
+    test('then the endpoints positional parameter has correct type', () {
+      var type = endpointDefinitions
+          .firstOrNull?.methods.firstOrNull?.parametersNamed.firstOrNull?.type;
+      expect(type?.className, 'String');
+    });
+
+    test('then the endpoint methods named parameter is required', () {
+      var required = endpointDefinitions.firstOrNull?.methods.firstOrNull
+          ?.parametersNamed.firstOrNull?.required;
+      expect(required, isTrue);
+    });
+
+    test('then the endpoint methods named parameter has dart parameter', () {
+      var dartParameter = endpointDefinitions.firstOrNull?.methods.firstOrNull
+          ?.parametersNamed.firstOrNull?.dartParameter;
+      expect(dartParameter, isNotNull);
+    });
+  });
+
+  group('Given valid endpoint with private method when analyzed', () {
+    var collector = CodeGenerationCollector();
+    var testDirectory =
+        Directory(path.join(testProjectDirectory.path, const Uuid().v4()));
+
+    late List<EndpointDefinition> endpointDefinitions;
+    late EndpointsAnalyzer analyzer;
+    setUpAll(() async {
+      var endpointFile = File(path.join(testDirectory.path, 'endpoint.dart'));
+      endpointFile.createSync(recursive: true);
+      endpointFile.writeAsStringSync('''
+import 'package:serverpod/serverpod.dart';
+
+class ExampleEndpoint extends Endpoint {
+  Future<String> _hello(Session session, String name) async {
+    return 'Hello \$name';
+  }
+}
+''');
```

Comment:
I would say it should stay here since the endpoint is still valid. We just don't have any methods defined for it. But we wills till generated an endpoint definition as asserted by the test.